---
layout: page
title: About Me
sidebar_link: true
---

Hi, there! I'm Gibraan Rahman, but everyone calls me Gibs.
I'm a third-year bioinformatics PhD student in the [Knight Lab](https://knightlab.ucsd.edu/) at UCSD.
My primary research interest is computational microbiology - focusing on cancer-microbiome interactions and statistical methods.
In my spare time, I like to collect vinyl, sing karaoke, and watch basketball (go Rockets!).

Contact me at: gibsramen (at) gmail.com

(†) denotes entry held online due to COVID-19

## Education

### Bioinformatics & Systems Biology, PhD

University of California, San Diego (Aug 2018 - Present)

### Biomedical Engineering, BS

Concentration in computational biology

University of Texas at Austin (August 2014 - May 2018)

---

## Teaching

### Graduate Teaching Assistant - Network Biology and Biomedicine<sup>†</sup>

University of California, San Diego (March 2020 - June 2020)

* Assisted with planning and organizing lectures
* Established infrastructure for remote teaching, distribution of materials, & grading
* Designed and graded homework assignments, exams, and final project
    * Developed both mathematical & computational problems dealing with network biology
    * Integrated published datasets into computational problems
* Held virtual office hours for students to clarify material and receive assistance

### "Software Engineering on a Team" Workshop Co-Host<sup>†</sup>

Bioinformatics & Systems Biology Incoming Student Bootcamp

University of California, San Diego (September 2020)

* Presented slide deck of best practices for collaborative software development
    * Unit testing
    * Style
    * Git/GitHub
    * Continuous integration
* Co-organized interactive demo of above skills involving student participation
    * Helped >20 students modify code and submit pull requests to GitHub
* Answered student questions on collaborative code writing

---

## Skills

### Programming

I am most comfortable with **Python** for analysis, tool development, and general purpose scripting.
Occasionally I utilize tools written in **R** for bioinformatic analysis although I prefer Python.
For command line scripting I often develop **bash** scripts though I am still learning to properly use `cut`, `awk`, etc.
Finally, I have recently begun implementing **Stan** into my workflows as I have been increasingly interested in Bayesian statistics.

### Software

I am very comfortable on the **UNIX/Linux** command line for both development and analysis.
When writing code, I almost always make use of **Git** & **GitHub** for version control and open source.
I integrate these tools with **unit testing** and **continuous integration** through Travis or GitHub actions.
I am a strong proponent of test-driven development and robust software testing.

In my spare time I have also started learning front-end technologies - primarily **JavaScript & TypeScript through React.js** with **HTML** and **CSS**.
I've also taken to learning about the **AWS** ecosystem of products with experience in **S3**, **Lambda**, **API Gateway**, **CloudFront**, & **EC2**.
These skills are still a bit crude but I hope to keep working on them in my personal projects.

### Data Analysis

I have strengths in **pandas** and **Numpy** analysis for general data analysis.
When tasked with machine learning problems or modelling I typically use **sklearn** and/or **statsmodels**.
Additionally, I am very passionate and competent in data visualization through **Matplotlib** and **Seaborn**.
Recently I have also been learning **Dask** and **xarray** for parallelization and organization of multidimensional data.
For microbiome specific analysis I use and develop for the **QIIME 2** ecosystem of tools.

---

## Presentations

### "Acceleration of Bioinformatics Workloads" - CRISP 2020<sup>†</sup>

Gave presentation about computational improvements to metagenomics processing workflows.

---

## Awards

### Cockrell School of Engineering Honors Scholarship

University of Texas at Austin (2014 - 2018)

This scholarship is awarded to incoming first-years to the Cockrell School of Engineering at UT Austin and renewed each year contingent upon academic performance.

---

## Open Source Contributions

* [Qurro](https://github.com/biocore/qurro) (Developer)
* [EMPress](https://github.com/biocore/empress) (Developer)
